File: AdminPanel/AdminPanel.Web/Modules/ServerTypes/AdhomStoreCom/CategoryForm.ts
import { StringEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface CategoryForm {
    Name: StringEditor;
    Description: StringEditor;
    Image: StringEditor;
}

export class CategoryForm extends PrefixedContext {
    static formKey = 'AdhomStoreCom.Category';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CategoryForm.init)  {
            CategoryForm.init = true;

            var w0 = StringEditor;

            initFormType(CategoryForm, [
                'Name', w0,
                'Description', w0,
                'Image', w0
            ]);
        }
    }
}

File: AdminPanel/AdminPanel.Web/Modules/AdhomStoreCom/Category/CategoryForm.cs
using Serenity.ComponentModel;
using Serenity.Web;

namespace AdminPanel.AdhomStoreCom.Forms
{
    [FormScript("AdhomStoreCom.Category")]
    [BasedOnRow(typeof(CategoryRow), CheckNames = true)]
    public class CategoryForm
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public string Image { get; set; }
    }
}